stdlib/websub/src/main/ballerina/websub/Module.md

```diff
@@ -10,40 +10,86 @@ This implementation supports introducing all WebSub components: subscribers, pub
  hub.
    
 4. Publisher notifies the hub of updates to the topic, and the content to deliver is identified.
 
 5. The hub delivers the identified content to the subscribers of the topic.
 
-#### Features
+### Features
+
+#### Subscriber
+
 This module allows introducing a WebSub Subscriber Service with `onIntentVerification`, which accepts HTTP GET requests for intent verification, and `onNotification`, which accepts HTTP POST requests for notifications. The WebSub Subscriber Service provides the following capabilities:
  - Subscription Requests are sent at service start time for the hub and topic, which are either specified as annotations or discovered based on the resource URL specified as an annotation.
  - Auto Intent Verification against the topic specified as an annotation, or discovered based on the resource URL specified as an annotation, if `onIntentVerification` is not specified.
  - Signature Validation for authenticated content distribution if a secret is specified for the subscription.
   
+#### Hub
+
 A WebSub compliant hub based on the Ballerina Message Broker is also available for use as a remote hub or to be used by publishers who want to have their own internal hub. Ballerina's WebSub Hub honors specified lease periods and supports authenticated content distribution.
 
+##### Enabling Data Persistence for the Hub
+
+The Ballerina WebSub Hub supports persistence of topic and subscription data that needs to be restored when the hub is 
+restarted.
+ 
+Users can introduce their own persistence implementation, by introducing an object type structurally 
+equivalent to the `websub:HubPersistenceStore` abstract object. Alternatively, the H2 based 
+`websub:H2HubPersistenceStore` object made available in the `ballerina/websub` module could be used to persist data.
+
+Persistence can be enabled by setting a suitable `websub:HubPersistenceStore` value for the `hubPersistenceStore` field 
+in the `HubConfiguration` record passed to the `websub:startHub()` function.
+
+```ballerina
+import ballerina/h2;
+import ballerina/http;
+import ballerina/websub;
+
+// Define an `h2:Client` to use with the `websub:H2HubPersistenceStore`.
+h2:Client h2Client = new({
+    path: "./target/hubDB",
+    name: "hubdb",
+    username: "user1",
+    password: "pass1"
+});
+
+// Define a `websub:H2HubPersistenceStore` as a `websub:HubPersistenceStore`.
+websub:HubPersistenceStore hubPersistenceStore = new websub:H2HubPersistenceStore(h2Client);
+// Set the defined persistence store as the `hubPersistenceStore` in the `hubConfig` record.
+websub:HubConfiguration hubConfig = {
+    hubPersistenceStore: hubPersistenceStore
+};
+
+// Pass the `hubConfig` record when starting up the hub to enable persistence.
+var result = websub:startHub(new http:Listener(8080), hubConfiguration = hubConfig);
+```
+
+Any subscriptions added at the hub will now be available even when the hub is restarted.
```

Comment:
```suggestion
Any subscriptions added to the hub will now be available even when the hub is restarted.
```

Reply:
Again, the hub is more like a separate component. You subscribe to topics at the hub, not to the hub itself.